packages/nextra-theme-docs/src/components/search.tsx

```diff
@@ -241,85 +148,95 @@ export function Search({
     [handleChange]
   )
 
-  return (
-    <div className={cn('nextra-search _relative md:_w-64', className)}>
-      {renderList && (
-        <div className="_fixed _inset-0 _z-10" onClick={() => setShow(false)} />
-      )}
+  const handleSelect = useCallback(
+    async (searchResult: SearchResult | null) => {
+      if (!searchResult) return
+      // Calling before navigation so selector `html:not(:has(*:focus))` in styles.css will work,
+      // and we'll have padding top since input is not focused
+      inputRef.current?.blur()
+      await router.push(searchResult.route)
+      // Clear input after navigation completes
+      setMenu(false)
+      onChange('')
+    },
+    [router, setMenu, onChange]
+  )
 
-      <Input
-        ref={input}
+  // const [selected, setSelected] = useState<SearchResult | null>(null)
+  //
+  // useEffect(() => {
+  //   setSelected(results[0] || null)
+  // }, [results])
+
+  return (
+    <Combobox
+      as="div"
+      // value={selected}
+      onChange={handleSelect}
+      className={cn('nextra-search _relative md:_w-64', className)}
+    >
+      <ComboboxInput
+        ref={inputRef}
+        as={Input}
+        autoComplete="off"
+        value={value}
         onCompositionStart={handleCompositionStart}
         onCompositionEnd={handleCompositionEnd}
```

Comment:
need to remove `onCompositionStart` and `onCompositionEnd` because the previous usage was for uncontrolled input. (I've tested it and the issue in #2833 does not reoccur.)

Reply:
removed, anything else to improve on this PR? or ready to merge?